app/concepts/position.rb

```diff
@@ -0,0 +1,18 @@
+class Position
+  attr_reader :stock, :total_quantity, :market_price, :cost
+
+  def initialize(stock:, quantity:, market_price:, cost:)
+    @stock = stock
+    @total_quantity = quantity
+    @market_price = market_price
+    @cost = cost
+  end
+
+  def total_value
+    (total_quantity || 0) * (market_price || 0)
```

Comment:
Is this because `total_quantity` and `market_price` might be `nil`? If so, you could consider casting them `to_d` in your initializer. `nil.to_d # => 0`